src/Features/Core/Portable/IntroduceVariable/AbstractIntroduceParameterService.cs

```diff
@@ -0,0 +1,688 @@
+            IMethodSymbol methodSymbol, string newMethodIdentifier, SyntaxEditor editor, CancellationToken cancellationToken)
+        {
+            var document = semanticDocument.Document;
+            var syntaxFacts = semanticDocument.Document.GetRequiredLanguageService<ISyntaxFactsService>();
+            var oldMethodDeclaration = expression.FirstAncestorOrSelf<SyntaxNode>(node => IsContainedInParameterizedDeclaration(node), true)!;
+            var returnExpression = new List<SyntaxNode>
+            {
+                editor.Generator.ReturnStatement(expression.WithoutTrailingTrivia())
+            };
+
+            var typeSymbol = semanticDocument.SemanticModel.GetTypeInfo(expression, cancellationToken).Type;
+            var newMethod = CodeGenerationSymbolFactory.CreateMethodSymbol(methodSymbol, name: newMethodIdentifier, returnType: typeSymbol);
+
+            var newMethodDeclaration = editor.Generator.MethodDeclaration(newMethod, returnExpression);
+            editor.InsertBefore(oldMethodDeclaration, newMethodDeclaration);
+        }
+
+        /// <summary>
+        /// Generates a method declaration containing a call to the method that introduced the parameter
+        /// </summary>
+        /// 
+        private void GenerateNewMethodOverload(SemanticDocument semanticDocument, TExpressionSyntax expression,
+            IMethodSymbol methodSymbol, SyntaxEditor editor)
+        {
+            var document = semanticDocument.Document;
+            var syntaxFacts = semanticDocument.Document.GetRequiredLanguageService<ISyntaxFactsService>();
+            var generator = editor.Generator;
+            var oldMethodDeclaration = expression.FirstAncestorOrSelf<SyntaxNode>(node => IsContainedInParameterizedDeclaration(node), true)!;
+            var arguments = generator.CreateArguments(methodSymbol.Parameters);
+            arguments = AddExpressionArgumentToArgumentList(arguments, expression.WithoutTrailingTrivia(), generator);
+            var memberName = methodSymbol.IsGenericMethod
+                ? generator.GenericName(methodSymbol.Name, methodSymbol.TypeArguments)
+                : generator.IdentifierName(methodSymbol.Name);
+
+            var invocation = generator.InvocationExpression(memberName, arguments);
+            List<SyntaxNode> invocationReturn;
+
+            if (methodSymbol.ReturnsVoid)
+            {
+                invocationReturn = new List<SyntaxNode>
+                {
+                    invocation
+                };
+            }
+            else
+            {
+                invocationReturn = new List<SyntaxNode>
+                {
+                    generator.ReturnStatement(invocation)
+                };
+            }
+
+            var newMethod = CodeGenerationSymbolFactory.CreateMethodSymbol(methodSymbol);
+            var newMethodDeclaration = generator.MethodDeclaration(newMethod, invocationReturn);
+            editor.InsertBefore(oldMethodDeclaration, newMethodDeclaration);
```

Comment:
please add tests where there use has set the preference to use expression bodied members, and ensure the right thing is generated.

Reply:
It reverts them to normal method declarations, is that okay?